### ISSUE TITLE
Focus states broken on internal header

### ISSUE THREAD

**Issue by rmccar** [ONSdigital/design-system#3263]
Focus states aren't correctly displaying on internal headers, this would be an accessibility issue for users with impaired sight because there isn't enough contrast. The logo should be black with a yellow background when focused and the link should have black text with black underline when focused like other logos and links are styled.

To reproduce:
Navigate to the [internal header](https://service-manual.ons.gov.uk/design-system/components/header) example and focus on the ons logo or sign out link (using a full size screen)

<img width="393" alt="Screenshot 2024-07-11 at 14 26 43" src="https://github.com/ONSdigital/design-system/assets/42928680/63c9b012-5882-4c8e-9a4a-e4d0f310acb4">
<img width="478" alt="Screenshot 2024-07-11 at 14 26 34" src="https://github.com/ONSdigital/design-system/assets/42928680/783ea21b-a52e-45b3-afcb-f4b2cdf83814">

**Comment by e-livingstone** [comment]
Discuss with @Majsol
